src/test/java/org/aszjch/demoapp/domain/article/port/KafkaArticlePublisherIT.java

```diff
@@ -0,0 +1,38 @@
+package org.aszjch.demoapp.domain.article.port;
+
+import org.aszjch.demoapp.TestcontainersConfiguration;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Import;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
+@Import(TestcontainersConfiguration.class)
+class KafkaArticlePublisherIT {
+
+    @Autowired
+    private ArticlePublisher kafkaArticlePublisher;
+    @SpyBean
+    private KafkaTestListener kafkaTestListener;
+    @Captor
```

Comment:
Hmmm ciekawe użycie Captor'a. Być może jest to sensowne podejście :) zobaczymy jak się sprawdzi.

Reply:
Właśnie tak sobie badam trochę te captory ostatnio i np. jak chcę sprawdzić wysyłkę po rest na jakiś system, to stawiam sobie controller w testach i łapię argument captorem to co do niego przyjdzie. Wtedy mogę sobie łatwo zobaczyć, czy json na 200% poprawnie się sformatował, czy tak jak tutaj, mogę sprawdzić, czy po kafce przyszła wiadomość poprawnie.

Najbardziej do mnie przemawia w takim podejściu to, że nie muszę pchać dodatkowych technologii, jak jakieś RestAssured, czy inny MockWebServer.